### PR TITLE
Introduce security-schemas instead of using authorization header in the API requests

### PR DESCRIPTION
**Pull request by ayeshLK**
Part of: [#6087](https://github.com/ballerina-platform/ballerina-library/issues/6087)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: ballerina/types.bal [hunk 1]
```diff
@@ -76,6 +78,13 @@ public type ProxyConfig record {|
     string password = "";
 |};
 
+# OAuth2 Client Credentials Grant Configs
+public type OAuth2ClientCredentialsGrantConfig record {|
+    *http:OAuth2ClientCredentialsGrantConfig;
+    # Token URL
+    string tokenUrl = "/oauth/token";
```
Comment (lnash94) [inline ballerina/types.bal:85]: Is it possible to access tokenURL via this relative path?

Comment (ayeshLK) [reply ballerina/types.bal:85]: As per the OpenAPI spec [1] this is doable if token-endpoint is also accessible via API base URL. It seems like the OpenAPI tool does not support this. I will create an issue for this.

[1] - [https://swagger.io/docs/specification/authentication/oauth2/](https://swagger.io/docs/specification/authentication/oauth2/)

Comment (ayeshLK) [reply ballerina/types.bal:85]: Fixed with following:
- [334978dd9a8af5a894cdea770acf4b63c3c0b1d0](https://github.com/ballerina-platform/module-ballerinax-zuora/pull/4/commits/334978dd9a8af5a894cdea770acf4b63c3c0b1d0)
- [eff6c3b665175be1c2394eaa0e1f1b334c79614c](https://github.com/ballerina-platform/module-ballerinax-zuora/pull/4/commits/eff6c3b665175be1c2394eaa0e1f1b334c79614c)

Comment (ayeshLK) [reply ballerina/types.bal:85]: Created following issue [1] to track this

[1] - [https://github.com/ballerina-platform/ballerina-library/issues/6369](https://github.com/ballerina-platform/ballerina-library/issues/6369)

**Pull request closed without merging**